aiida/manage/configuration/settings.py

```diff
@@ -38,7 +38,7 @@ def create_instance_directories():
     """
     from aiida.common import ConfigurationError
 
-    directory_base = pathlib.Path(AIIDA_CONFIG_FOLDER).expanduser()
+    directory_base = AIIDA_CONFIG_FOLDER.expanduser()
```

Comment:
AIIDA_CONFIG_FOLDER is already typed at the top as `pathlib.Path`

Reply:
Technically it can also be `None` so this would except. But then again the original `pathlib.Path(None)` would also except. Guess this is a potential bug lying in the wait. Fine to keep this change for now